### PR TITLE
User can view list of nearby restaurants

### PR DESCRIPTION
**Pull request by lcegit**
# PT Link: https://www.pivotaltracker.com/story/show/154171740

# Description:
Add only restaurants close to the user location and in the same city

# What we learned:
* How to understand legacy code
* How to use properly JSON files and possible alternative solutions
* How to use `let` in unit testing
* How to keep step definitions and files organised
* How to fix previous features and steps to make tests pass
* How to update the attribute of a model by associating it to another model
* How to test on Chrome with debugger
* How to use Selenium to remove headless and test extensively on Chrome
* How to properly translate from JavaScript ES6 to ES5
* How to save data in a cookie and reload a page to use the saved cookie
* How to create and save new fixtures files for webmock
* How to use Google Maps API to update attribute values while using the Geocoder gem
* Making use of javascript promises, variables that are set and saved to cookies by reloading the browser

# Screenshot:
screen shot 2018-01-19 at 18 19 29

### CODE REVIEW

**Review by tochman — DISMISSED** [review]
God job

**Review by SophieSjoberg — COMMENTED** [review]
Wow, great job you guys! 👍

**Comment by tochman** [comment]
Please fix conflicts

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -19,5 +7,62 @@
...
+function redirectWithLocation(position) {
+  var url = new URL(window.location.href);
+
+  if (document.body.dataset.geocoded != 'true') {
+    window.location.replace(url);
+    document.body.dataset.geocoded = true;
+  }
+}
+
+setCookie = function setCookie(name, value) {
```
Comment (tochman) [inline app/assets/javascripts/application.js]: This repeat the function name. Please follow the standard set in other functions

**Review by tochman — CHANGES_REQUESTED** [review]
One more issue need to be addressed

File: app/assets/javascripts/application.js (hunk 1 above)
Comment (NegativeKarma) [reply app/assets/javascripts/application.js]: Done!

**Review by tochman — APPROVED** [review]
Good job

**Pull request merged**